### PR TITLE
Use external print, gc and assert handlers in nuttx-stm32f4 target

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: targets/nuttx-stm32f4/jerry_main.c [hunk 1]
```diff
@@ -447,58 +409,66 @@ int jerry_main (int argc, char *argv[])
...
-
-    jerry_run_simple ((jerry_char_t *) source_p, strlen (source_p), flags);
-    return JERRY_STANDALONE_EXIT_CODE_OK;
-  }
-
   jerry_init (flags);
 
-  register_js_function ("assert", assert_handler);
-  register_js_function ("gc", gc_handler);
+  register_js_function ("assert", jerryx_handler_assert);
```
Comment (akosthekiss) [inline targets/nuttx-stm32f4/jerry_main.c]: why not reuse `jerryx_handler_register_global` (completely, or in a wrapped form as in jerry-main)?

Comment (robertsipka) [inline targets/nuttx-stm32f4/jerry_main.c]: @akosthekiss : You're right, I missed it. I've updated it accordingly.

File: targets/nuttx-stm32f4/jerry_main.c [hunk 2]
```diff
@@ -447,58 +384,66 @@ int jerry_main (int argc, char *argv[])
...
-    return JERRY_STANDALONE_EXIT_CODE_OK;
-  }
-
   jerry_init (flags);
 
-  register_js_function ("assert", assert_handler);
-  register_js_function ("gc", gc_handler);
+  jerryx_handler_register_global ("assert", jerryx_handler_assert);
+  jerryx_handler_register_global ("gc", jerryx_handler_gc);
+  jerryx_handler_register_global ("print", jerryx_handler_print);
```
Comment (akosthekiss) [inline targets/nuttx-stm32f4/jerry_main.c]: the value returned by `jerryx_handler_register_global` must be freed by `jerry_release_value` (that's why jerry-main has kept `register_js_function` as a wrapper around `jerryx_handler_register_global`, to ensure that such extra steps are performed, see https://github.com/jerryscript-project/jerryscript/blob/master/jerry-main/main-unix.c#L313)

**Comment by robertsipka** [comment]
@akosthekiss : Thanks, patch updated.

**Pull request merged**